File: src/main/java/io/junseok/mealmate/domain/restaurantmenu/presentation/RestaurantMenuController.java
package io.junseok.mealmate.domain.restaurantmenu.presentation;

import io.junseok.mealmate.domain.restaurantmenu.dto.request.MenuRegister;
import io.junseok.mealmate.domain.restaurantmenu.service.RestaurantMenuService;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/restaurant-menu")
@CrossOrigin
public class RestaurantMenuController {
    private final RestaurantMenuService restaurantMenuService;

    @PostMapping("/{restaurantId}")
    public ResponseEntity<Void> createMenu(
        @PathVariable Long restaurantId,
        @RequestBody List<MenuRegister> menuList){
        restaurantMenuService.registerMenu(restaurantId,menuList);
        return ResponseEntity.ok().build();
    }
}

File: src/main/java/io/junseok/mealmate/domain/restaurantmenu/presentation/RestaurantMenuController.kt
package io.junseok.mealmate.domain.restaurantmenu.presentation

import io.junseok.mealmate.domain.restaurantmenu.dto.request.MenuRegister
import io.junseok.mealmate.domain.restaurantmenu.service.RestaurantMenuService
import org.springframework.http.ResponseEntity
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.*
import java.security.Principal

@RestController
@RequestMapping("/api/restaurant-menu")
@CrossOrigin
class RestaurantMenuController(
    private val restaurantMenuService: RestaurantMenuService
) {
    @PostMapping("/{restaurantId}")
    @PreAuthorize("hasRole('ROLE_ADMIN')")
    fun createMenu(
        @PathVariable restaurantId: Long,
        @RequestBody menuList: List<MenuRegister>
    ): ResponseEntity<Unit> {
        restaurantMenuService.registerMenu(restaurantId, menuList)
        return ResponseEntity.ok().build()
    }

    @PatchMapping("/{restaurantMenuId}")
    @PreAuthorize("hasRole('ROLE_ADMIN')")
    fun modifyMenu(
        @PathVariable restaurantMenuId: Long,
        @RequestBody menuRegister: MenuRegister,
        principal: Principal
    ): ResponseEntity<Unit> {
        restaurantMenuService.updateMenu(
            restaurantMenuId,
            menuRegister,
            principal.name
        )
        return ResponseEntity.ok().build()
    }

    @DeleteMapping("/{restaurantMenuId}")
    @PreAuthorize("hasRole('ROLE_ADMIN')")
    fun removeMenu(@PathVariable restaurantMenuId: Long, principal: Principal):ResponseEntity<Unit>{
        restaurantMenuService.deleteMenu(restaurantMenuId)
        return ResponseEntity.ok().build()
    }
}
